File: data_structures.py
class Stack(object):
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        item_to_pop = self.items[-1]
        self.items = self.items[:-1]
        return item_to_pop

class Queue(object):
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def next(self):
        next_item = self.items[0]
        self.items = self.items[1:]
        return next_item

File: data_structures.js
function Stack() {
  return {
    items: [],
    push: function(item) {
      this.items.push(item);
    },
    pop: function() {
      return this.items.pop();
    }
  }
}

function Queue() {
  return {
    items: [],
    push: function(item) {
      this.items.push(item);
    },
    next: function() {
      return this.items.shift();
    }
  }
}
